client/src/components/ShoppingLists.js

```diff
@@ -24,37 +24,74 @@ const useStyles = makeStyles((theme) => ({
 
 
 
-const ShoppingLists = ()=>{
+const ShoppingLists = (props)=>{
     const classes = useStyles();
     const shListsContext = useContext(ShListsContext);
+    const [userLists, setUserLists] = useState(null);
+
+    const getUserLists = async () => {
+		try {
+			const res = await axios.get('/users/'+props.userName);
+            await setUserLists(res.data[0].my_lists);
+		} catch (error) {
+			console.log('Error getting user');
+		}
+	};
 
     useEffect(() => {
+        if(props.userName !== undefined) {
+            getUserLists();
+        } else {
+            setUserLists(null);
+        }
         window.scrollTo(0, 0);
     }, []);
 
```

Comment:
When you use an empty array, the `[]`, as a second parameter for `useEffect`, useEffect will be run only once when this component mounts.https://blog.carbonfive.com/replacing-component-lifecycle-methods-with-react-hooks/ This may be the reason why after you get the first user's lists, you can't get second, third ... user's lists.

Reply:
Oh woww!! Thanks a lot @ChianHuei That works like a charm! Problem solved!

Reply:
Oh noes But it goes in infinite loop. It keeps calling the function again and again!

Reply:
yeah, it's easy to cause endless loop. Therefore, using `useEffect` to update context or state is not a good choice. The better timing to trigger context-updating is on-event, such as clicking on a button/link,whick triggers it once and then stop. But often, we don't have other option besides `useEffect`. so, set flags.

Reply:
Problem solved! Thanks again @ChianHuei Slack - 3rd solution worked after removing ternary operators and using if...else structure.!